networkx/algorithms/centrality/second_order_centrality.py

```diff
@@ -106,6 +109,20 @@ def second_order_centrality(G):
         raise nx.NetworkXException("Empty graph.")
     if not nx.is_connected(G):
         raise nx.NetworkXException("Non connected graph.")
+    if [True for (i, j, d) in G.edges(data=True)
```

Comment:
The ```any``` function does this without creatingg a list -- not a big deal, but probably easier to read too.
 
    if any(d.get('weight', 0) < 0 for u, v, d in G.edges(data=True)):
        raise(...)